File: my-solutions/0150-evaluate-reverse-polish-notation/solution.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        evalStack = deque()
        def performOp(op1, op2, op):
            if op == "+":
                return op1 + op2
            elif op == "-":
                return op1 - op2
            elif op == "*":
                return op1 * op2
            else:
                # print(op1, op2, int(op1/op2))
                return int(op1/op2)
        for i in range(len(tokens)):
            if tokens[i] in ["+", "-", "*", "/"]:
                op1 = int(evalStack[-1])
                evalStack.pop()
                op2 = int(evalStack[-1])
                evalStack.pop()
                evalStack.append(performOp(op2, op1, tokens[i]))
            else:
                evalStack.append(int(tokens[i]))
            # print(evalStack)

        return evalStack[-1]

            

File: my-solutions/0150-evaluate-reverse-polish-notation/solution.cpp

#include<string>

class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<string> operands;
        unordered_set<string> operators({"+", "-", "/", "*"});
        for(int i = 0; i < size(tokens); i++) {
            if(operators.find(tokens[i]) != operators.end()) {
                int op1 = stoi(operands.top());
                operands.pop();
                int op2 = stoi(operands.top());
                operands.pop();
                string operation = *(operators.find(tokens[i]));
                int result = 0;
                switch(operation[0]) {
                    case '+': result = op2 + op1; break;
                    case '-': result = op2 - op1; break;
                    case '*': result = op2 * op1; break;
                    case '/': result = op2 / op1; break;
                    default: cout << "Abort!" << endl;
                }
                operands.push(to_string(result));
            } else {
                operands.push(string(tokens[i]));
            }
        }

        return stoi(operands.top());
    }
};
